app/components/filters/select-multiple-filter.ts

```diff
@@ -33,22 +35,20 @@ export default class SelectMultipleFilterComponent extends FilterComponent<Signa
 
   @action
   async inserted() {
-    const searchField = this.args.searchField;
-
     if (!this.router.currentRoute) {
       console.error('Current route is not available');
       return;
     }
 
-    const flattenedHaystack: Option[] = [];
-    const haystack = (await this.args.options) as unknown as GroupedOptions[];
-    if (haystack?.[0]?.['groupName']) {
-      haystack.forEach((group) => {
-        group['options'].forEach((option: Option) => {
-          flattenedHaystack.push(option);
-        });
-      });
-    }
+    const searchField = this.args.searchField;
+    const haystack = await this.args.options;
+    const flattenedHaystack = haystack.reduce((acc, value) => {
+      if (Array.isArray(value.options)) {
+        return [...acc, ...value.options];
+      } else {
+        return [...acc, value as Option];
+      }
+    }, [] as Option[]);
 
     const results = deserializeArray(this.args.queryParam).flatMap(
```

Comment:
make it so the value gets split on ",". This way when multiple classifications get selected they'll get split up properly.

```ts
const results = deserializeArray(this.args.queryParam).flatMap(
      (queryParam) => {
        if (!queryParam) return [];
        const queryParamValue = this.getQueryParam(queryParam);
        const values = queryParamValue ? deserializeArray(queryParamValue) : [];
        return values
          .flatMap((value) => value.split(','))
          .map((value) => {
            return flattenedHaystack.find(
              (option) =>
                get(option, searchField) === value &&
                option['type'] === queryParam
            );
          })
          .filter(Boolean) as Option[];
      }
    );
```

Reply:
Other than that it looks very good! :)

Reply:
@snenenenenenene The problem was that we mix serialization separator. We used `+` on locations and `,` on governingBodyClassification. I refactor the `selectChange` method in order to apply `+` everywhere, that fix the bug too.